packages/strapi-plugin-rest-cache/server/register.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/**
+ * @typedef {import('@strapi/strapi').Strapi} Strapi
+ */
+const chalk = require('chalk');
+const debug = require('debug');
+
+const { CacheProvider } = require('./types');
+const { resolveUserStrategy } = require('./utils/config/resolveUserStrategy');
+const { injectMiddlewares } = require('./utils/middlewares/injectMiddlewares');
+
+const createProvider = async (providerConfig, { strapi }) => {
```

Comment:
This needs to be moved back to bootstrap currently it will try to run this before the provider exists:

![image](https://github.com/strapi-community/strapi-plugin-rest-cache/assets/8593673/943adca2-077f-4664-ab82-2dc490191647)


We can still register the middlewares but we can't create the provider until we get to the bootstrap.